service-pdf/src/Opg/Lpa/Pdf/AbstractLp1.php

```diff
@@ -732,6 +872,58 @@ private function getTrustAttorney(array $attorneys)
         return null;
     }
 
+    private function generateBarcodeAsImage()
+    {
+        // Generate the barcode
+        $renderer = Barcode::factory(
+            'code39',
+            'image',
+            [
+                'text' => str_replace(' ', '', $this->formattedLpaRef),
+                'drawText' => false,
+                'factor' => 1,
+                'barHeight' => 25
+            ],
+        );
+
+        // Create a PNG with the barcode only
+        $barcodeOnlyPng = $renderer->draw();
+
+        $barcodePngFile = $this->getIntermediatePdfFilePath('barcode.png');
+        imagepng($barcodeOnlyPng, $barcodePngFile);
+
+        // Create a PDF from the PNG; we need this as Pdftk can't stamp
+        // a PDF with anything other than another PDF
+        $pdf = new TCPDF('P', 'pt');
+
+        // set empty margins
+        $pdf->SetMargins(0, 0, 0);
+        $pdf->SetHeaderMargin(0);
+        $pdf->SetFooterMargin(0);
+
+        // remove header and footer borders
+        $pdf->SetPrintHeader(false);
+        $pdf->SetPrintFooter(false);
+
+        // set auto page breaks
+        $pdf->SetAutoPageBreak(false, 0);
+
+        // set image scale factor
+        $pdf->setImageScale(1);
+
+        $pdf->AddPage();
+
+        // filename, x position, y position, width, height, type, link, align, resize, dpi
+        $pdf->Image($barcodePngFile, 40, 789);
+
+        $barcodePdfFile = $this->getIntermediatePdfFilePath('barcode.pdf');
```

Comment:
Re-check getIntermediatePdfFilePath returns a path unique to this PDF.

Reply:
This is used in the old code for the generated barcode PDF, so we aren't doing anything new or unusual here.